api/routes/sshkeys.go

```diff
@@ -128,3 +129,24 @@ func CreatePrivateKey(c apicontext.Context) error {
 
 	return c.JSON(http.StatusOK, privKey)
 }
+
+func EvaluateKeyHostname(c apicontext.Context) error {
+	svc := sshkeys.NewService(c.Store())
+	var device models.Device
+
+	pubKey, err := svc.GetPublicKey(c.Ctx(), c.Param("fingerprint"), c.Param("tenant"))
+	if err != nil {
+		return c.JSON(http.StatusForbidden, err)
+	}
+
+	if err := c.Bind(&device); err != nil {
+		return c.JSON(http.StatusForbidden, err)
+	}
+
+	ok, err := svc.EvaluateKeyHostname(c.Ctx(), pubKey, device)
```

Comment:
Check `err` and take `ok` as default `return` if all previous checks work.

Reply:
this way?